bundles/org.palladiosimulator.editors.sirius.ui/src/org/palladiosimulator/editors/sirius/ui/wizard/project/NewPalladioProjectWizard.java

```diff
@@ -214,49 +218,80 @@ private void createAndOpenProject(final IProjectDescription description, final I
     private void handleTemplate(final IProject projectHandle, final SubMonitor subMonitor) throws CoreException {
         final AT selectedTemplate = this.palladioTemplatePage.getSelectedTemplate();
         if (selectedTemplate != null) {
-            addToProject(computeTemplatePath(selectedTemplate), projectHandle, subMonitor);
+            TemplateInitiationContext context = new TemplateInitiationContext(projectHandle, selectedTemplate);
+            context.addToProject(computeTemplatePath(selectedTemplate), projectHandle, subMonitor);
         }
     }
 
-    private void addToProject(final URI path, final IContainer target, final SubMonitor subMonitor)
-            throws CoreException {
-        for (final File source : FileHelper.getFiles(path.toString())) {
-            final IPath newTarget = new Path(source.getName());
-            if (source.isDirectory()) {
-                addFolderToProject(path, source, target.getFolder(newTarget), subMonitor);
-            } else {
-                addFileToProject(source, target.getFile(newTarget), subMonitor);
+    /**
+     * Encapsulates one invocation of the template loading.
+     */
+    private static class TemplateInitiationContext {
+        public final IProject projectHandle;
```

Comment:
Fields are not accessed from outside. Changing them to private would improve encapsulation.